### PR TITLE
add code contribution guidelines

### PR DESCRIPTION
**Pull request by hanars**
Add guidance for external code contributions to our README

### CODE REVIEW

**Review by sjahl — APPROVED** [review]
I think this looks great!

A suggestion for future thought (I don't think we need to block merging on this): A lot of the contributor guide(line)s that I like tend to include instructions on how to set up a development environment that resembles the way we work. For us, that might include instructions on how to set up a virtualenv, and how to run tests locally, etc. This kind of guide is also useful for our own internal hires.

Some good examples are:

- https://hail.is/docs/0.2/getting_started_developing.html
- https://matrix-org.github.io/synapse/latest/development/contributing_guide.html (something like this is way overkill for us, but kind of shows the length at which one could go :) )

**Comment by hanars** [comment]
> A suggestion for future thought (I don't think we need to block merging on this): A lot of the contributor guide(line)s that I like tend to include instructions on how to set up a development environment that resembles the way we work.

Yeah, so we actually don't have any documentation anywhere for how to set up seqr for development 😬 , which is definitely a huge gap (I just spent a very painful few hours trying to help Avery do it from my memory). I made a ticket to write this documentation so I don't keep forgetting

File: README.md [hunk 1]
```diff
@@ -29,3 +29,32 @@ On-prem installs can be created using docker-compose:
...
+[discussion forum](https://github.com/broadinstitute/seqr/discussions)
+
+To contribute code:
+
+* Submit a GitHub pull request against the master branch.
+* Break your work into small, single-purpose patches whenever possible. 
+However, do not break apart features to the point that they are not functional 
+(i.e. updates that require changes to both front end and backend code should be submitted as a single change)
+* For larger features, add a detailed description to the pull request to explain the changes and your approach
+* Make sure that your code passes all our tests and style linting
```
Comment (ShifaSZ) [inline README.md:54]: Do the tests refer to those automated by GitHub? More specific descriptions are preferred.

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (hanars) [reply README.md:54]: this is intentionally vague, it is up to the contributor if they want to set up tests to run locally or rely on our automated github tests, depending on what kind of change they are contributing

**Review by ShifaSZ — APPROVED** [review]

**Pull request merged**